backfill/2024-11-06-metrics-pings-metric-labels-over-max-length/README.md

```diff
@@ -0,0 +1,224 @@
+
+On 2024-11-04, a [fix](https://bugzilla.mozilla.org/attachment.cgi?id=9435189) landed to update the JSON schemas to allow for up to 71 character in metric label names.
+
+See [Bug 1929078](https://bugzilla.mozilla.org/show_bug.cgi?id=1929078) for more information.
+
+The steps to backfill are:
+1. Run the affected pings through the decoder dataflow job, writing to staging tables
+2. Dedupe the staging tables 
+3. Insert into the stable tables (requires SRE assistance)
+
+## Affected Pings
+
+To find the affected ping volume and dates, we can look in `moz-fx-data-shared-prod.monitoring.payload_bytes_error_structured`:
+
+```sql
+SELECT
+  DATE(submission_timestamp) AS d,
+  COUNT(*),
+FROM
+  `moz-fx-data-shared-prod.monitoring.payload_bytes_error_structured`
+WHERE
+  DATE(submission_timestamp) BETWEEN "2024-10-25" AND "2024-11-06"
+  AND
+    error_message LIKE 'org.everit.json.schema.ValidationException: #/metrics/% expected maxLength: 61, actual: %'
+GROUP BY d
+ORDER BY d
+```
+
+Only `metrics` pings are affected, as confirmed by the following query:
+
+```sql
+SELECT
+  document_type,
+  COUNT(*),
+FROM
+  `moz-fx-data-shared-prod.monitoring.payload_bytes_error_structured`
+WHERE
+  DATE(submission_timestamp) BETWEEN "2024-10-25" AND "2024-11-06"
+  AND error_message LIKE 'org.everit.json.schema.ValidationException: #/metrics/% expected maxLength: 61, actual: %'
+GROUP BY
+  document_type
+ORDER BY
+  2 DESC
+```
+
+Only, `firefox-desktop` is affected:
+
+```sql
+SELECT
+  document_namespace,
+  COUNT(*),
+FROM
+  `moz-fx-data-shared-prod.monitoring.payload_bytes_error_structured`
+WHERE
+  DATE(submission_timestamp) BETWEEN "2024-10-25" AND "2024-11-06"
+  AND error_message LIKE 'org.everit.json.schema.ValidationException: #/metrics/% expected maxLength: 61, actual: %'
+GROUP BY
+  document_namespace
+ORDER BY
+  2 DESC
+
+-- The 11 org-mozilla-firefox pings that show up are all unrelated to this specific issue and exceed the label length by much more:
+--
+-- SELECT
+--   document_namespace,
+--   error_message
+-- FROM
+--   `moz-fx-data-shared-prod.monitoring.payload_bytes_error_structured`
+-- WHERE
+--   DATE(submission_timestamp) BETWEEN "2024-10-25" AND "2024-11-06"
+--   AND error_message LIKE 'org.everit.json.schema.ValidationException: #/metrics/% expected maxLength: 61, actual: %'
+--   AND document_namespace = "org-mozilla-firefox"
+-- ORDER BY
+--   2 DESC
+```
+
+## Setting up the Backfill Project
+
+To grant dataset creation permissions and to get access to `payload_bytes_error`: https://github.com/mozilla-services/cloudops-infra/pull/6055
```

Comment:
While you are granted dataset create permissions by this process it shouldn't be necessary to create any datasets because they will have already been created by the backfill config. It's expected after that config is in place that all you need to do is point dataflow at the right input and output tables.